txircd/modules/extra/chanhistory.py

```diff
@@ -0,0 +1,78 @@
+from twisted.plugin import IPlugin
+from txircd.config import ConfigValidationError
+from txircd.module_interface import IMode, IModuleData, Mode, ModuleData
+from txircd.utils import ModeType, now
+from zope.interface import implements
+from datetime import timedelta
+from weakref import WeakKeyDictionary
+
+class ChanHistory(ModuleData, Mode):
+	implements(IPlugin, IModuleData, IMode)
+
+	name = "ChannelHistory"
+	affectedActions = {
+		"commandextra-PRIVMSG": 1
```

Comment:
I would expect this to be switched with the "join" action. That, or put both actions here.